CMakeLists.txt

```diff
@@ -12,7 +12,10 @@ include(CheckCXXCompilerFlag)
 
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)  # Falling back to different standard it not allowed.
-set(CMAKE_CXX_EXTENSIONS NO)  # Make sure no compiler-specific features are used.
+set(CMAKE_CXX_EXTENSIONS OFF)  # Make sure no compiler-specific features are used.
+
+# Build edlib as static library by default
+set(BUILD_SHARED_LIBS OFF CACHE BOOL "Build all libraries as shared")
```

Comment:
So now, edlib is built only as static library, while before I was building is as both shared and static, right? Why is this better, are there any negative sides, how did you make this choice?

Reply:
The problem is, creating two differently named targets makes flipping between them at configure time tricky, because you need to change target names. [`BUILD_SHARED_LIBS`](https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html) is a vanilla CMake variable that many packagers use to switch between shared and static libraries. Notice also how changing this required changing AppVeyor, because previously the `-DBUILD_SHARED_LIBS=ON` had zero effect, whereas now it caused MSVC to fail, due to the Windows loader not finding the DLLs (because they aren't in the `PATH`).

In general, you should always prefer vanilla CMake solutions to working around it, because it makes packaging for distributions so much easier.

Reply:
Ok, I have to admit I am certainly missing some knowledge here to properly understand all implications, but I am getting that this way we use that variable, which is standard, to decide if we are building shared or static lib, and that makes our and life of others easier in certain cases since the name of the lib is always the same and this behaviour is more standard. Sounds good!

Reply:
Yes, that is correct ✅